interfaces/IBF-dashboard/src/app/services/map.service.spec.ts

```diff
@@ -20,9 +29,57 @@ describe('MapService', () => {
       ],
     });
     service = TestBed.inject(MapService);
+    adminLevelService = TestBed.inject(AdminLevelService);
   });
 
   it('should be created', () => {
     expect(service).toBeTruthy();
   });
+
+  describe('getLayerByName', () => {
+    beforeEach(() => {
+      service.layers = MOCK_LAYERS;
+    });
+
+    it('should get a layer from its name', () => {
+      const layerName = IbfLayerName.waterpointsInternal;
+
+      const expected = MOCK_LAYERS[0];
+
+      expect(service.getLayerByName(layerName)).toEqual(expected);
+    });
+
+    it('should be undefined if no layer has the provided name', () => {
+      const layerName = IbfLayerName.gauges;
+
+      expect(service.getLayerByName(layerName)).toBeUndefined();
+    });
+  });
+
+  describe('getPlaceCodeParent', () => {
+    it('should return the palceCodeParent of the provided placeCode', () => {
+      // eslint-disable-next-line @typescript-eslint/dot-notation
+      adminLevelService['countryDisasterSettings'] =
+        MOCK_COUNTRYDISASTERSETTINGS;
+      // eslint-disable-next-line @typescript-eslint/dot-notation
+      adminLevelService['adminLevel'] = 3;
+
+      expect(service.getPlaceCodeParent(MOCK_PLACECODE)).toEqual(
+        MOCK_PLACECODE.placeCodeParent.placeCode,
+      );
+    });
+  });
+
+  describe('getAreaByPlaceCode', () => {
+    it('should return the triggered area corresponding to the provided placeCode or placeCodeParent', () => {
+      service.triggeredAreas = MOCK_TRIGGEREDAREAS;
+
+      expect(
+        service.getAreaByPlaceCode(
+          MOCK_PLACECODE.placeCode,
+          MOCK_PLACECODE.placeCodeParent.placeCode,
+        ),
+      ).toEqual(MOCK_TRIGGEREDAREAS[0]);
```

Comment:
It doesn't feel intuitive why this is expected to equal. Maybe add 'const expectedArea = MOCK_TRIGGEREDAREAS[0]', or something else to make it more clear.

Also, I feel you're mixing up the use of MOCK_PLACECODE a bit. I think mock in this context is really meant to be used for defining responses (or intermediate data, like service.triggeredAreas), and not for input. You don't need to mock the input right? You can simply pass the input.. So it's mostly semantics, but I feel like it's important to split.. So just say 'const placeCode = abc' and pass that in the method.